dask_expr/_collection.py

```diff
@@ -896,8 +896,24 @@ def shuffle(
                 raise TypeError(
                     "index must be aligned with the DataFrame to use as shuffle index."
                 )
-        elif pd.api.types.is_list_like(on) and not is_dask_collection(on):
-            on = list(on)
+        else:
+            if pd.api.types.is_list_like(on) and not is_dask_collection(on):
+                on = list(on)
+            elif isinstance(on, str):
+                on = [on]
+            bad_cols = [
+                index_col
+                for index_col in on
+                if (index_col not in self.columns) and (index_col != self.index.name)
+            ]
+            if len(bad_cols) == 1:
```

Comment:
I think you can reformat this into a single statement, like:

f"Cannot shuffle on {bad_cols}, column(s) not in dataframe to shuffle"